contrib/coredns/policy/policy.go

```diff
@@ -196,3 +201,29 @@ func (p *policyPlugin) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dn
 	status = dns.RcodeServerFailure
 	return dns.RcodeSuccess, errInvalidAction
 }
+
+// checkOwnIPEndpointDebug checks if query made to configured ownIPEndpoint and is debug, then returns the status accordingly
+func (p *policyPlugin) checkOwnIPEndpointDebug(w dns.ResponseWriter, r *dns.Msg) (int, bool) {
+	if strings.HasSuffix(p.conf.ownIPEndpoint, p.conf.debugSuffix) &&
```

Comment:
don't check this - you guaranteed this in config code

Reply:
The feature requires that debug queries to `ownIPEndPoint` should get responded with client's IP.  Corefile could have only `ownIPEndpoint` and no debug suffix. Since we use the same variable to concat the debug prefix to it, we could end up with only ownIPEndpoint and no debug prefix, in which case a non-debug query will also respond with client's IP if we omit this check here.

Reply:
This check doesn't depend on query itself, so no need to check it each time. You can reset `ownIPEndpoint` to "" at config time in case if some config options are incorrect. The check `r.Question[0].Name == p.conf.ownIPEndpoint` will fail if `ownIPEndpoint == ""`, since query name cannot be empty